src/tray/unix/SDL_tray.c

```diff
@@ -0,0 +1,597 @@
+    entry->parent = menu;
+
+    if (flags & SDL_TRAYENTRY_CHECKBOX) {
+        entry->item = gtk_check_menu_item_new_with_label(label);
+        gtk_check_menu_item_set_active(GTK_CHECK_MENU_ITEM(entry->item), !!(flags & SDL_TRAYENTRY_CHECKED));
+    } else {
+        entry->item = gtk_menu_item_new_with_label(label);
+    }
+
+    gtk_widget_set_sensitive(entry->item, !(flags & SDL_TRAYENTRY_DISABLED));
+
+    SDL_TrayEntry **new_entries = (SDL_TrayEntry **) SDL_realloc(menu->entries, (menu->nEntries + 1) * sizeof(SDL_TrayEntry *));
+
+    if (!new_entries) {
+        SDL_free(entry);
+        return NULL;
+    }
+
+    menu->entries = new_entries;
+    menu->nEntries++;
+
+    for (int i = menu->nEntries - 1; i > pos; i--) {
+        menu->entries[i] = menu->entries[i - 1];
+    }
+
+    new_entries[pos] = entry;
+
+    gtk_widget_show(entry->item);
+    gtk_menu_shell_insert(menu->menu, entry->item, (pos == menu->nEntries) ? -1 : pos);
+
+    return entry;
+}
+
+SDL_TrayEntry *SDL_AppendTrayEntry(SDL_TrayMenu *menu, const char *label, SDL_TrayEntryFlags flags)
+{
+    SDL_TrayEntry *entry = SDL_malloc(sizeof(SDL_TrayEntry));
+
+    if (!entry) {
+        return NULL;
+    }
+
+    SDL_memset((void *) entry, 0, sizeof(*entry));
+    entry->parent = menu;
+
+    if (flags & SDL_TRAYENTRY_CHECKBOX) {
+        entry->item = gtk_check_menu_item_new_with_label(label);
+        gtk_check_menu_item_set_active(GTK_CHECK_MENU_ITEM(entry->item), !!(flags & SDL_TRAYENTRY_CHECKED));
+    } else {
+        entry->item = gtk_menu_item_new_with_label(label);
+    }
+
+    gtk_widget_set_sensitive(entry->item, !(flags & SDL_TRAYENTRY_DISABLED));
+
+    SDL_TrayEntry **new_entries = (SDL_TrayEntry **) SDL_realloc(menu->entries, (menu->nEntries + 1) * sizeof(SDL_TrayEntry *));
+
+    if (!new_entries) {
+        SDL_free(entry);
+        return NULL;
+    }
+
+    new_entries[menu->nEntries] = entry;
+    menu->entries = new_entries;
+    menu->nEntries++;
+
+    gtk_widget_show(entry->item);
+    gtk_menu_shell_append(menu->menu, entry->item);
+
+    return entry;
+}
+
+void SDL_AppendTraySeparator(SDL_TrayMenu *menu)
+{
+    GtkWidget *item = gtk_separator_menu_item_new();
+    gtk_widget_show(item);
+    gtk_menu_shell_append(menu->menu, item);
+}
+
+void SDL_SetTrayEntryChecked(SDL_TrayEntry *entry, bool checked)
+{
+    SDL_Unsupported();
+}
+
+bool SDL_GetTrayEntryChecked(SDL_TrayEntry *entry)
+{
+    SDL_Unsupported();
+    return false;
+}
+
+void SDL_SetTrayEntryEnabled(SDL_TrayEntry *entry, bool enabled)
+{
+    SDL_Unsupported();
+}
+
+bool SDL_GetTrayEntryEnabled(SDL_TrayEntry *entry)
+{
+    SDL_Unsupported();
+    return false;
+}
+
+void SDL_SetTrayEntryCallback(SDL_TrayEntry *entry, SDL_TrayCallback callback, void *userdata)
+{
+    entry->callback = callback;
+    entry->userdata = userdata;
+
+    g_signal_connect(entry->item, "activate", G_CALLBACK(call_callback), entry);
+}
+
+void SDL_DestroyTray(SDL_Tray *tray)
```

Comment:
Destory functions (maybe others too) should be no-ops when you pass in a null pointer.